### ISSUE TITLE
Performance drops dramatically changing tracking state from limited to normal

### ISSUE THREAD

**Issue by JacoGasp** [ProjectDent/ARKit-CoreLocation#236]
I pulled the last commits from develop and now I always experience really poor performance when the tracking state change from `normal` to `limited, insufficient feature` to `normal` again.
Everything becomes extremely laggy and I have to restart my application.

Here the log that it prompts every time

```
camera did change tracking state: limited, initializing
camera did change tracking state: limited, relocalizing
camera did change tracking state: normal
camera did change tracking state: limited, insufficient features
camera did change tracking state: normal
[Technique] World tracking performance is being affected by resource constraints [1]
[Technique] VIO error callback: 26859.984836, 1, Frame processing rate has fallen below pre-set threshold
[Technique] VIO error callback: 26860.484814, 1, Frame processing rate has fallen below pre-set threshold
[Technique] World tracking performance is being affected by resource constraints [1]
[Technique] VIO error callback: 26861.251447, 1, Frame processing rate has fallen below pre-set threshold
[Technique] VIO error callback: 26861.534768, 1, Frame processing rate has fallen below pre-set threshold
[Technique] VIO error callback: 26861.634764, 1, Frame processing rate has fallen below pre-set threshold
[Technique] World tracking performance is being affected by resource constraints [1]
...
```

**Comment by pcmanik** [comment]
@intere can you please take a look on that?
Currently it's in unusable state.

**Comment by halmueller** [comment]
If someone who's seeing the performance regression will run a git bisect and pin down which commit introduced the regression, I'll take a look (https://flaviocopes.com/git-bisect/ is a good quickstart).

**Comment by pcmanik** [comment]
It was introduced between 1.1.0 and 1.2.0.

**Comment by halmueller** [comment]
You two are in the best positions of anyone to nail this down, since you have a reproducible regression case. No one else can run the diagnostics. Do you have a repro that you can share? 

To pin down what's going on, I suggest:

- run the `git bisect` to confirm that you've found the commit. I don't see anything in https://github.com/ProjectDent/ARKit-CoreLocation/commit/bdf0b1d2d52ae4210b02d50d3531c38a0e18f2ec that looks suspicious; it's mostly a red commit with a change to convenience initializer.
- What frame rates are you seeing before/after? 
- What does Instruments have to say about the functions that are using the most CPU?

**Comment by JacoGasp** [comment]
From what I've seen so far, the issue _seems_ disappeared, without any significant changes in my code. So I suppose that the issue could potentially be  related to a iOS bug that has been fixed with one of the latest releases.

But I can say that:

- iOS was showing 60 fps even if everything (even the UI response) was extremely laggy. It was looking like the app was freezing every ~1s
- With the Time Profiler instrument I saw that the function using the most CPU was for sure `locationNode.updatePositionAndScale()` at [L267]( https://github.com/ProjectDent/ARKit-CoreLocation/blob/develop/Sources/ARKit-CoreLocation/SceneLocationView.swift#L267)
